src/resource/solidDataset.test.ts

```diff
@@ -788,6 +788,61 @@ describe("saveSolidDatasetAt", () => {
       ).toBeGreaterThan(0);
     });
 
+    it("ignores hash fragments in the target IRI if any", async () => {
+      const mockedResponse = new Response();
+      jest
+        .spyOn(mockedResponse, "url", "get")
+        .mockReturnValue("https://some.url");
+      const mockFetch = jest
+        .fn(window.fetch)
+        .mockReturnValue(Promise.resolve(mockedResponse));
+
+      const mockDataset = getMockUpdatedDataset(
+        {
+          additions: [
+            DataFactory.quad(
+              DataFactory.namedNode("https://some.vocab/subject"),
+              DataFactory.namedNode("https://some.vocab/predicate"),
+              DataFactory.namedNode("https://some.vocab/object"),
+              undefined
+            ),
+          ],
+          deletions: [
+            DataFactory.quad(
+              DataFactory.namedNode("https://some-other.vocab/subject"),
+              DataFactory.namedNode("https://some-other.vocab/predicate"),
+              DataFactory.namedNode("https://some-other.vocab/object"),
+              undefined
+            ),
+          ],
+        },
+        "https://some.pod/resource"
+      );
+
+      await saveSolidDatasetAt(
+        "https://some.pod/resource#something",
+        mockDataset,
+        {
+          fetch: mockFetch,
+        }
+      );
+
+      expect(mockFetch.mock.calls).toHaveLength(1);
+      expect(mockFetch.mock.calls[0][0]).toEqual(
+        "https://some.pod/resource#something"
```

Comment:
So in line with my other suggestion, this would become

```suggestion
        "https://some.pod/resource"
```

Reply:
Well in that case I think it's up to the `fetch` function to strip the hash fragment from the target IRI before making the request, as opposed to the client app tampering with the target IRI. However, in line with your previous suggestion, the library should use the URL of the response (here likely to be `https://some.pod/resource`) instead of the user-provided `https://some.pod/resource#something` target IRI. That's what's enforced in f4f2d817579809804f5c6db76ea63438550d7d4d, and tested in c020d3ed48efc11504b4e4ce6866b6fdfebb061c.